template.yaml

```diff
@@ -109,71 +111,71 @@ Resources:
         SSEEnabled: true
 
   # API declaration
-  TodoApi:
-    Type: AWS::Serverless::Api
+  TodoApi: # this is implictly defined based on what functions we push - if we want to be explicit create a swagger definition of api
+    Type: AWS::Serverless::Api # api gateway instance
     MethodSettings: 
         DataTraceEnabled: true
         MetricsEnabled: true
         HttpMethod: '*'
-        ResourcePath: !Sub '${VersionParam}/*'
-        LoggingLevel: INFO
+        ResourcePath: !Sub '${VersionParam}/*' # !sub is substitute - https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-sub.html
+        LoggingLevel: INFO 
     AccessLogSetting:
       DestinationArn: !Sub 'arn:${AWS::Partition}:logs:${AWS::Region}:${AWS::AccountId}:log-group:${ApiAccessLogGroup}'
       Format: '$context.identity.sourceIp $context.authorizer.claims.sub [$context.requestTime] "$context.httpMethod $context.resourcePath $context.protocol" $context.status $context.requestId $context.awsEndpointRequestId $context.xrayTraceId $context.responseLatency $context.integrationLatency "$context.error.message"'
     Properties:
-      Name: TodoApi
-      StageName: !Ref StageNameParam
+      Name: TodoApi # change
+      StageName: !Ref StageNameParam 
       TracingEnabled: true
       Cors:
         AllowOrigin: "'*'"
         AllowMethods: "'OPTIONS,HEAD,GET,PUT,POST,DELETE'"
         AllowHeaders: "'Content-Type,X-Amz-Date,Authorization,X-Api-Key,X-Amz-Security-Token'"
-      Auth:
+      Auth: #  - https://github.com/aws/serverless-application-model/blob/master/versions/2016-10-31.md#api-auth-object
         Authorizers:
           CognitoAuthorizer:
-            UserPoolArn: !GetAtt "TodoUserPool.Arn"
+            UserPoolArn: !GetAtt "TodoUserPool.Arn" # update this to link correct userpool id
```

Comment:
Just change this name for now as well, still need to look into what our link would be; should be similar regardless.

Reply:
oh its what our user pool name is -> like above we instantiate the user pool called todouserpool - our name should be TakeNotesPool so the value is TakeNotesPool.Arn which gets the amazon resource number which links it